### DISCUSSION TITLE
Reusing a graphql filter DTO in a rest controller DTO.

### DISCUSSION THREAD

**Discussion by abrenoch** [TriPSs/nestjs-query#318] [Q&A]
Hey there, we are looking for the best way to support the same filter pattern that a graphql DTO uses in a REST controller.

Accessing and utilizing the query builder is easy enough:
```
    const filter = {
      catalogId: {
        in: catalogs.map((c) => c.id),
      },
    };
    const fqb = new FilterQueryBuilder<AssetEntity>(this.assetRepository);
    const s: Query<AssetEntity> = {
      filter,
    };
    const q = fqb.select(s).limit(100).offset(0);
    const assets = await q.getMany();
```

But we are having a hard time working out how the REST controller should be decorated in order to properly validate the type usage.

Something like this feels close:
```
  @ClassField(() => Filter)
  readonly filter: Filter<AssetDTO>;
```

But `Filter` does not seem to have a class constructor, so using it in the `@ClassField` decorator is a no-go. The field could of course be accepted as a string and parsed into an object, but I feel like that isn't taking advantage of the validation options at hand.

Or maybe I'm just way off base here?

Thanks!

**Comment by TriPSs** [comment]
Currently the lib has no support for REST, this is being added though in #200, it currently still in development (I'm developing it inside a other project and porting it back here) but still needs a bit of work (documentation, tests, use correct naming etc)
